src/pages/Promotions/List/index.tsx

```diff
@@ -0,0 +1,190 @@
+      ...(activeTab === "past" ? { endDate: { before: formattedToday } } : {})
+    }
+  }, {
+    keepPreviousData: true,
+    select: (response) => {
+      const promotions = filterNodes(response.promotions.nodes).map((promotion) => ({
+        ...promotion,
+        promotionType: promotion.promotionType as PromotionType,
+        actions: filterNodes(promotion.actions)
+      }));
+      return {
+        totalCount: response.promotions.totalCount,
+        promotions
+      };
+    }
+  });
+
+  const columns = useMemo((): ColumnDef<Promotion>[] => [
+    {
+      id: "select",
+      header: ({ table }) => (
+        <Checkbox
+          inputProps={{ "aria-label": "select all promotions" }}
+          disableRipple
+          {...{
+            checked: table.getIsAllRowsSelected(),
+            indeterminate: table.getIsSomeRowsSelected(),
+            onChange: table.getToggleAllRowsSelectedHandler()
+          }}
+        />
+      ),
+      cell: ({ row }) => (
+        <Checkbox
+          disableRipple
+          inputProps={{ "aria-label": "select promotion" }}
+          {...{
+            checked: row.getIsSelected(),
+            indeterminate: row.getIsSomeSelected(),
+            onChange: row.getToggleSelectedHandler()
+          }}
+        />
+      ),
+      meta: {
+        padding: "checkbox"
+      }
+    },
+    {
+      accessorKey: "name",
+      header: "Name"
+    },
+    {
+      accessorKey: "referenceId",
+      header: "ID"
+    },
+    {
+      accessorKey: "promotionType",
+      header: "Promotion Type",
+      cell: (row) =>
+        <Typography variant="body2">{PROMOTION_TYPE_OPTIONS[row.getValue<PromotionType>()]?.label || "Unknown"}</Typography>
+    },
+    {
+      accessorFn: (row) => row.actions?.[0]?.actionParameters?.discountCalculationType,
+      header: "Action",
+      id: "actionType",
+      enableSorting: false,
+      cell: (row) =>
+        <Typography variant="body2">{CALCULATION_TYPE_OPTIONS[row.getValue<CalculationType>()]?.label || "Noop"}</Typography>
+    },
+    {
+      accessorKey: "startDate",
+      header: "Start Date",
+      cell: ({ getValue }) => <Typography variant="body2">{formatDate(new Date(getValue()))}</Typography>
+    },
+    {
+      accessorKey: "endDate",
+      header: "End Date",
+      cell: ({ getValue }) =>
+        <Typography variant="body2">{getValue() ? formatDate(new Date(getValue())) : "--"}</Typography>
+    },
+    {
+      id: "status",
+      header: "Status",
+      cell: ({ row }) => <StatusChip promotion={row.original}/>,
+      enableSorting: false,
+      meta: {
+        align: "right"
+      }
+    }
+  ], []);
+
+  const handleChangeTab = (value: string) => {
+    setSearchParams({ type: value });
+    onRowSelectionChange({});
+  };
+
+  const disabledActionItem = Object.keys(rowSelection).length === 0;
+  return (
+    <Container sx={{ padding: "20px 30px" }} maxWidth={false}>
```

Comment:
I would prefer to not have absolute pixel values. The x-axis padding is already defined by the Container component so no need to set it, and for the y-axis we can use the theme spacing API by setting it to 3 for example.